dynamic_rest/routers.py

```diff
@@ -180,6 +182,24 @@ def register_resource(self, viewset, namespace=None):
             'viewset': viewset
         }
 
+        # Make sure the resource name isn't registered, either
+        # TODO: Think of a better way to clean this up, there's a lot of
```

Comment:
I added this `TODO` because I do think we could come up with a better way of doing this.  We're enforcing that both `resource_name` and `resource_key` be unique, and they should have a 1:1 mapping between the two, so I think we could come up with a way to reduce this to a single identifier.